File: 백준/Bronze/2884. 알람 시계/알람 시계.swift
let inputArray: [Int] = (readLine() ?? "0 0").split(separator: " ").map{ Int($0) ?? 0 }

// 0 <= H <= 23, 불필요한 0은 제외하고 출력
// 0 <= M <= 59, 불필요한 0은 제외하고 출력

let H: Int = inputArray[0]
let M: Int = inputArray[1]

// 조정 값을 저장 할 변수 선언 및 0으로 초기화
var calcH: Int = 0
var calcM: Int = 0

// 일찍 일어 날 시간 변수
let adjustM = 45

// 분 계산
if (M < adjustM) {
    // 일찍 일어 날 시간이 입력된 시간보다 크다면 시간 계산도 필요함
    calcH = -1
    calcM = 60 + M - adjustM
} else {
    // 그렇지 않다면 분 계산만 진행함
    calcM = M - adjustM
}

// 시간 계산
if (calcH == -1) {
    // 시간 계산이 필요한 경우 날짜 변경이 필요한 지 확인함
    if (H == 0) {
        // 날짜 변경이 필요하다면 23시로 변경함
        calcH = 23
    } else {
        // 그렇지 않다면 입력된 시간에 -1을 더함
        calcH += H
    }
} else {
    // 시간 계산이 필요 없다면 입력된 시간을 사용
    calcH = H
}

print(String(calcH), String(calcM))

File: 백준/Bronze/2884. 알람 시계/알람 시계.py
# 0 <= H <= 23, 0H와 같이 불필요한 0은 생략
# 0 <= M <= 59, 0M와 같이 불필요한 0은 생략
H, M = map(int, input().split())

# 계산을 위해 조정 값 선언 및 0으로 초기화
calcH = 0
calcM = 0

# 분 계산
if (M < 45) :
    calcM = 60 + M - 45
    calcH = -1
else :
    calcM = M - 45
    
# 시간 조정이 필요한 경우
if (calcH == -1) :
    # 날짜가 조정되는 경우
    if (H == 0) :
        calcH = 23
    # 1시간 - 되는 경우
    else :
        calcH += H
# 시간 조정이 필요없는 경우
else :
    calcH = H
    
print(str(calcH) + " " + str(calcM))